Getting-Started/Code/Source-Control/index.md

```diff
@@ -0,0 +1,139 @@
+
+#### Macros
+
+Macros can be implemented in four different ways, and you do want to source control your changes to Macro implementations, so track the files in the following locations:
+
+- **Partial View Macros** - stored in /views/macropartials as .cshtml files
+- **Dynamic Razor** - stored in /macroScripts as .cshtml files
+- **UserControls** - stored in /userControls folder as .ascx and ascx.cs files
+- **XSLT** - stored in /xslt folder as .xslt files
+- 
+#### Controllers/Classes/Custom Code
+
+Any supporting custom code for your application should be in version control, eg any
+- C# implementation,
+  - Surface Controllers, 
+  - RenderMVCControllers,  
+  - ViewModels, 
+  - Helpers / Extension Methods, 
+  - Services etc
+- Supporting class library projects, 
+- App_Code, 
+- Models generated by Modelsbuilder in API or App_Data mode
+
+...all should be added to source/version control.
+
+#### Config
+
+Your site's /config folder contains the set of configuration files for your Umbraco site. Add these to source control along with config.transforms to change variables for different environments during deployment.
+
+#### DocumentType - Backoffice Structure Changes
+
+When you create and edit DocumentTypes, MediaTypes, DataTypes, Macros, DictionaryItems, Languages, MemberTypes & Templates('names' not the actual files) in the Umbraco Backoffice these values are stored in the Umbraco Database, making them very difficult to source control in a 'file based' version control system.
+
+However there is a third party package called *uSync* - https://our.umbraco.org/projects/developer-tools/usync/ which when installed will serialize these settings to disk, in a folder called /uSync - enabling you to version control these changes too, and import and apply to other Umbraco environments - read settings carefully for automatically importing changes - Also see [uSync Snapshots](https://our.umbraco.com/packages/developer-tools/usyncsnapshots/) for managing a release of a'set' of changes between environments.
+
+If you are working with continuous integration, then *Chauffeur*, is another package worth investigating - https://github.com/aaronpowell/Chauffeur for issuing command line deliverables against an Umbraco install.
+
+**Note**
+It's possible to use [Umbraco Courier](https://our.umbraco.com/packages/umbraco-pro/umbraco-courier/) (Umbraco Cloud uses Umbraco Deploy) to try to move these kinds of changes through your different Umbraco development/staging/production environments, and potentially [source control 'Courier Revision Sets' for team development](http://stream.umbraco.org/video/2198253/team-development-with-courier)
+
+## Example gitignore
```

Comment:
maybe add: 

If you are working with GIT, you can add a gitignore file to specify intentionally untracked files that Git should ignore.